src/psyclone/psyad/domain/lfric/lfric_adjoint.py

```diff
@@ -116,16 +117,35 @@ def generate_lfric_adjoint(tl_psyir, active_variables):
     # help fix this problem as it would only be arguments that would
     # need to have the same names.
 
+    ctr_table = ad_container.symbol_table
+
+    # Before we can rename any routines, we need to know if they're referenced
+    # by any GenericInterfaceSymbols. Construct a reverse dict so that, given
+    # a RoutineSymbol, we can lookup which GenericInterfaceSymbol(s) refer
+    # to it.
+    interfaced_routines = dict()
+    for gisym in ctr_table.symbols:
+        if isinstance(gisym, GenericInterfaceSymbol):
+            for routine in gisym.routines:
+                if routine.symbol in interfaced_routines:
+                    interfaced_routines[routine.symbol].append(gisym)
+                else:
+                    interfaced_routines[routine.symbol] = [gisym]
+
```

Comment:
Is this still needed? Now GenericInterfaceSymbols has a list of symbols, not names, so it may allow renaming?

Reply:
It is still needed because a Symbol's name is immutable. Therefore, if we have to rename a kernel, we have to create a new RoutineSymbol and replace the relevant entry in `GenericRoutineSymbol.routines` with it. (This is why we need the `replace` method that you ask about later.)

Reply:
But we have the symbol_table.rename_symbol method, that does consistent renaming by changing the _name and the dict key. In addition, it checks that the name is not used externally or in a CodeBlock that the replace functionality lacks.

I tried replacing from line 120 to 149 with simply:
```
    for routine in routines:
        # We need to re-name the kernel routine.
        kernel_sym = ctr_table.lookup(routine.name)
        adj_kernel_name = create_adjoint_name(routine.name)
        ctr_table.rename_symbol(kernel_sym, adj_kernel_name)
        routine.name = adj_kernel_name
```
and it passes the tests.

Can you do this change and remove the replace if this is right?

Reply:
Ah! Can't believe I missed that. I too find that all the tests pass. I've just triggered the integration tests to be on the safe side too.